code/modules/power/supermatter/supermatter.dm

```diff
@@ -128,59 +187,65 @@
 						mob.apply_effect(rads, IRRADIATE)
 
 			explode()
+			emergency_lighting(0)
 
 	//Ok, get the air from the turf
 	var/datum/gas_mixture/env = L.return_air()
 
 	//Remove gas from surrounding area
 	var/datum/gas_mixture/removed = env.remove(gasefficency * env.total_moles())
 
-	if(!removed || !removed.total_moles())
-		damage += max((power-1600)/10, 0)
-		power = min(power, 1600)
-		return 1
+	//ensure that damage doesn't increase too quickly due to super high temperatures resulting from no coolant, for example. We dont want the SM exploding before anyone can react.
+	//We want the cap to scale linearly with power (and explosion_point). Let's aim for a cap of 5 at power = 300 (based on testing, equals roughly 5% per SM alert announcement).
+	var/damage_inc_limit = (power/300)*(explosion_point/1000)*DAMAGE_RATE_LIMIT
 
-	damage_archived = damage
-	damage = max( damage + ( (removed.temperature - 800) / 150 ) , 0 )
-	//Ok, 100% oxygen atmosphere = best reaction
-	//Maxes out at 100% oxygen pressure
-	oxygen = max(min((removed.oxygen - (removed.nitrogen * NITROGEN_RETARDATION_FACTOR)) / MOLES_CELLSTANDARD, 1), 0)
+	if(!env || !removed || !removed.total_moles())
+		damage += max((power - 15*POWER_FACTOR)/10, 0)
+	else
+		damage_archived = damage
 
-	var/temp_factor = 50
+	damage = max(0, damage + between(-DAMAGE_RATE_LIMIT, (removed.temperature - CRITICAL_TEMPERATURE) / 150, damage_inc_limit))
 
-	if(oxygen > 0.8)
-		// with a perfect gas mix, make the power less based on heat
+	//Maxes out at 100% oxygen pressure
+	oxygen = Clamp((removed.oxygen - (removed.nitrogen * NITROGEN_RETARDATION_FACTOR)) / removed.total_moles(), 0, 1)
+
+	var/temp_factor
+	var/equilibrium_power
+	if (oxygen > 0.8)
+		//If chain reacting at oxygen == 1, we want the power at 800 K to stabilize at a power level of 400
+		equilibrium_power = 400
 		icon_state = "[base_icon_state]_glow"
 	else
-		// in normal mode, base the produced energy around the heat
-		temp_factor = 30
+		//If chain reacting at oxygen == 1, we want the power at 800 K to stabilize at a power level of 250
+		equilibrium_power = 250
 		icon_state = base_icon_state
 
-	power = max( (removed.temperature * temp_factor / T0C) * oxygen + power, 0) //Total laser power plus an overload
-
-	//We've generated power, now let's transfer it to the collectors for storing/usage
-	transfer_energy()
+	temp_factor = ( (equilibrium_power/DECAY_FACTOR)**3 )/800
```

Comment:
The spacing here is basically opposite of what it should be. `temp_factor = ((equilibrium_power / DECAY_FACTOR) ** 3) / 800`